src/main/java/com/aws/greengrass/device/iot/CertificateRegistry.java

```diff
@@ -80,16 +80,16 @@ private Optional<String> fetchActiveCertificateId(String certificatePem) {
         return iotAuthClient.getActiveCertificateId(certificatePem);
     }
 
+    /**
+     * Returns IoT Certificate ID associated locally for given certificate PEM.
+     *
+     * @param certificatePem Certificate PEM
+     * @return Certificate ID or empty optional
+     */
     private Optional<String> getAssociatedCertificateId(String certificatePem) {
-        try {
-            String certId = certificateHashToIdMap.get(Digest.calculate(certificatePem));
-            if (certId != null) {
-                return Optional.of(certId);
-            }
-        } catch (NoSuchAlgorithmException e) {
-            return Optional.empty();
-        }
-        return Optional.empty();
+        String certId =  Optional.ofNullable(getCertificateHash(certificatePem))
+                .map(certificateHashToIdMap::get).orElse(null);
+        return Optional.ofNullable(certId);
```

Comment:
why do you need `orElse` here? Can't you just return from the `.map` call directly?

Reply:
right, updated return from `.map()` itself.